portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/delete-data")
+public class DeleteDataServlet extends HttpServlet {
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query q = new Query("Comment").setKeysOnly();
+    PreparedQuery pq = this.datastore.prepare(q);
+    for (Entity entity : pq.asIterable()){
```

Comment:
Datastore::delete has an overload that takes an iterable. So I think this can be simplified to

this.datastore.delete(pq.AsIterable());

https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/DatastoreService.html#delete_2

Reply:
`Datastore.delete()` accepts `Iterable<Key>` but `pq.asIterable()` returns `Iterable<Entity>`, so I try to map `Iterable<Entity>` to `Iterable<Key>`. `this.datastore.delete(this.datastore.put(pq.asIterable()))` works because `put` essentially overwrite all entities in the datastore and return `List<Key>`; this is not efficient nor clear in my opinion. 

I couldn't seem to find a better off-the-shelf method that maps `Iterable<Entity>` to `Iterable<Key>`. `getKey()` works only for `Entity` objects but not in batch, so a for-loop is always necessary.